packages/dart_frog_auth/test/src/dart_frog_auth_test.dart

```diff
@@ -0,0 +1,206 @@
+// ignore_for_file: prefer_const_constructors
+import 'dart:io';
+
+import 'package:dart_frog/dart_frog.dart';
+import 'package:dart_frog_auth/dart_frog_auth.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:test/test.dart';
+
+class _MockRequestContext extends Mock implements RequestContext {}
+
+class _MockRequest extends Mock implements Request {}
+
+class User {
+  const User(this.id);
+  final String id;
+}
+
+void main() {
+  group('BasicAuth', () {
+    late RequestContext context;
+    late Request request;
+    User? user;
+
+    setUp(() {
+      context = _MockRequestContext();
+      request = _MockRequest();
+      when(() => context.provide<User>(any())).thenReturn(context);
+      when(() => request.headers).thenReturn({});
+      when(() => context.request).thenReturn(request);
+    });
+
+    test('returns 401 when no Authorization header is not present', () async {
+      final middleware = basicAuthentication<User>(
+        userFromCredentials: (_, __) async => user,
+      );
+      expect(
+        await middleware((_) async => Response())(context),
+        isA<Response>().having(
+          (r) => r.statusCode,
+          'statusCode',
+          HttpStatus.unauthorized,
+        ),
+      );
+    });
+
+    test(
+      'returns 401 when Authorization header is present but invalid',
```

Comment:
Nit: should we group those tests that start with "returns 401"?

Reply:
I feel that the grouping here should be around the context, which in this case is `when Authorization header ...`, `returns 401` is the outcome of that context, and we have only one possible outcome from that group, so I feel it is not worth to group them since each group would have a since test.